addon/webextension/background/startBackground.js

```diff
@@ -0,0 +1,94 @@
+/* globals browser, main, communication */
+/* This file handles browser.browser.onClicked, and loads the rest of the background
+   page in response to that event. */
+
+this.startBackground = (function() {
+  const backgroundScripts = [
+    "log.js",
+    "makeUuid.js",
+    "catcher.js",
+    "background/selectorLoader.js",
+    "background/communication.js",
+    "background/auth.js",
+    "background/senderror.js",
+    "build/raven.js",
+    "build/shot.js",
+    "background/analytics.js",
+    "background/deviceInfo.js",
+    "background/takeshot.js",
+    "background/main.js"
+  ];
+
+  browser.browserAction.onClicked.addListener((tab) => {
+    loadIfNecessary().then(() => {
+      main.onClicked(tab);
+    }).catch((error) => {
+      console.error("Error loading Screenshots:", error);
```

Comment:
Raven.js is one of the larger things to load, I think (96k). So reporting something isn't easy. Or we try to load it while reporting... but then we're in a weird circular state where for some reason loading scripts isn't working, so we try to load a script to report it. It's a bummer, but I'm not sure what else to do.

Though it would be easy to at least pop up an error notification, even without Sentry reporting. That might be better than removing the button, or simply doing nothing. Removing the button could be weird: the button will be there, the person *clicks* on it, and then it's gone! Haha, tricked you! 

I don't think there's a ton of errors that will actually result in a rejected promise either. Script errors do not – unfortunately we have no good way of catching those (before or after this patch). I think only a missing script will trigger `onerror` and hence reject.